src/MaxBTCERC20.sol

```diff
@@ -5,27 +5,69 @@
 //   - rename contract to MaxBTCERC20
 //   - set decimals() to 8
 //   - remove decimals() @dev docstring
-//   - replace config with dedicated ICS20 storage slot
+//   - add rate limiting
 //   - disable renounceOwnership()
 pragma solidity ^0.8.28;
 
-import { IMintableAndBurnable } from "./IMintableAndBurnable.sol";
-import { ERC20Upgradeable } from "@openzeppelin-upgradeable/token/ERC20/ERC20Upgradeable.sol";
-import { UUPSUpgradeable } from "@openzeppelin-contracts/proxy/utils/UUPSUpgradeable.sol";
-import { Ownable2StepUpgradeable } from "@openzeppelin-upgradeable/access/Ownable2StepUpgradeable.sol";
-import { StorageSlot } from "@openzeppelin/contracts/utils/StorageSlot.sol";
+import {IMintableAndBurnable} from "./IMintableAndBurnable.sol";
+import {
+    ERC20Upgradeable
+} from "@openzeppelin-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import {
+    UUPSUpgradeable
+} from "@openzeppelin-contracts/proxy/utils/UUPSUpgradeable.sol";
+import {
+    OwnableUpgradeable
+} from "@openzeppelin-upgradeable/access/OwnableUpgradeable.sol";
+import {StorageSlot} from "@openzeppelin-contracts/utils/StorageSlot.sol";
 
-contract MaxBTCERC20 is IMintableAndBurnable, UUPSUpgradeable, ERC20Upgradeable, Ownable2StepUpgradeable {
-    /// @notice Caller is not the ICS20 contract
+contract MaxBTCERC20 is
+    IMintableAndBurnable,
+    UUPSUpgradeable,
+    ERC20Upgradeable,
+    OwnableUpgradeable
+{
+    /// @notice Caller is not allowed
     /// @param caller The address of the caller
-    error CallerIsNotICS20(address caller);
+    error CallerIsNotAllowed(address caller);
 
-    /// @notice Provided ICS20 address is invalid
-    error InvalidICS20();
+    /// @notice Eureka rate limits have been exceeded
+    /// @param requested Requested amount of tokens to burn/mint
+    /// @param limit Allowed amount of tokens to burn/mint
+    error EurekaRateLimitsExceeded(uint256 requested, uint256 limit);
 
-    /// @notice ERC-7201 slot for the ICS20 contract address
-    /// @dev keccak256(abi.encode(uint256(keccak256("maxbtc.erc20.ics20")) - 1)) & ~bytes32(uint256(0xff))
-    bytes32 private constant ICS20_STORAGE_SLOT = 0xaa9b9403d129a09996409713bb21f8632c135ae1789678b7128d16411b23e500;
+    event ConfigUpdated(address indexed updater, address ics20, address core);
```

Comment:
maybe it's worth to make ics20 and core addresses indexed as well?

Reply:
To be honest, I am not sure if any field in this event needs to be indexed. What if just I remove indexed from this line?

Reply:
let's remove it then